demo/api/src/warnings/warning-checker.command.ts

```diff
@@ -42,19 +48,19 @@ export class WarningCheckerCommand extends CommandRunner {
             const queryBuilderLimit = 100;
             const baseQueryBuilder = this.entityManager.createQueryBuilder(className);
             baseQueryBuilder
-                .select(["id", ...rootBlockData.map(({ column }) => column)])
+                .select([`${data.primaryKey} as id`, ...rootBlockData.map(({ column }) => column)])
                 .from(tableName)
                 .limit(queryBuilderLimit);
-            let rootBlocks: Array<{ [key: string]: BlockData }> = [];
+            let rootBlocks: RootBlockData[] = [];
             let offset = 0;
 
             do {
                 const queryBuilder = baseQueryBuilder.clone();
                 queryBuilder.offset(offset);
-                rootBlocks = (await queryBuilder.getResult()) as Array<{ [key: string]: BlockData }>;
-
+                rootBlocks = (await queryBuilder.getResult()) as RootBlockData[];
                 for (const { column, block } of rootBlockData) {
                     for (const rootBlock of rootBlocks) {
+                        if (typeof rootBlock[column] === "string") continue;
```

Comment:
Why is this necessary?

Reply:
The check isn't strictly necessary; it's just there to prevent an error from being thrown. I'm unsure how to properly type it.

Currently, I have the following interface:
```
interface RootBlockData {
    id: string;
    [key: string]: BlockData | string;
}
```

Since id is explicitly typed as a string, I can't enforce that every other key is of type BlockData. Instead, I have to use [key: string]: BlockData | string.

i changed it like that currently: 480701126e4673c7f9368001df91f7a8edaa5a5b